spring-session/src/main/java/org/springframework/session/web/http/SessionRepositoryFilter.java

```diff
@@ -132,6 +136,22 @@ public void setServletContext(ServletContext servletContext) {
 		this.servletContext = servletContext;
 	}
 
+	private final class HttpRequestSessionContext implements SessionContext {
```

Comment:
Thanks for your comment. You are right that in many instances it is preferable to have a static class. However, this won't work since the `SessionRepositoryRequestWrapper` is non-static. The reasons `SessionRepositoryRequestWrapper` is non-static is because it needs a reference to the type argument of `SessionRepositoryFilter` to be type safe.